ludwig/features/base_feature.py

```diff
@@ -490,3 +492,12 @@ def prepare_decoder_inputs(
         )
 
         return feature_hidden
+
+    def flatten(self, df: pd.DataFrame) -> pd.DataFrame:
+        """ Converts the output of batch_predict to a 1D array. """
+        return df
+
+    def unflatten(self, df: dd.DataFrame) -> dd.DataFrame:
```

Comment:
We'll need to make Dask a core dependency for this type hint to work, as currently it's optional and this is on the critical path. I don't think that's a bad idea, though, as it's pretty lightweight.

Reply:
Another option we may also want to consider using the DataFrameEngine abstraction we use elsewhere to make this agnostic to different DataFrame implementations we may want to use in the future (like Modin): 

https://github.com/ludwig-ai/ludwig/blob/master/ludwig/data/dataframe/base.py#L21

Here's an example of how we use it:

https://github.com/ludwig-ai/ludwig/blob/master/ludwig/utils/strings_utils.py#L114

Reply:
I like the idea of a DataFrame abstraction, but one concern I have is that some methods of the specific DF implementation might not be represented in `DataFrameEngine`. E.g., in the case of a Dask DataFrame, the `apply` method isn't defined for `DataFrameEngine`. I'm not sure how much of an issue you think this is?

One mitigation is to also keep adding methods to `DataFrameEngine` as needed.

Reply:
That's true, but one way we work around this is to have a dummy implementation for Pandas that just does the simple thing:

https://github.com/ludwig-ai/ludwig/blob/master/ludwig/data/dataframe/pandas.py#L52

But I think this is fine for now, as it does make it more clear that the context is different (full DF vs shard).

Reply:
Great!